### PR TITLE
feat(form): added demos, section title, stack control group mod

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/3380

Adds a `pf-m-stack` variation to control groups (similar to `pf-m-inline`) that creates a stacking layout and space between children in form control groups.

Added section titles.

Updated and added forms from #3380.

### CODE REVIEW

File: src/patternfly/demos/Form/examples/BasicForms.md [hunk 1]
```diff
@@ -281,3 +263,96 @@ section: components
...
+        {{#> form-label form-label--attribute=(concat 'for="' form--id form-group--id '"') required="true"}}Root URL{{/form-label}}
+        {{> form-group-label-help}}
+      {{/form-group-label}}
+      {{#> form-group-control}}
+        {{#> form-control controlType="input" input="true" form-control--attribute=(concat 'required type="text" id="' form--id form-group--id '" name="' form--id form-group--id '"')}}{{/form-control}}
+      {{/form-group-control}}
+    {{/form-group}}
+    {{#> form-group form-group--id="-uris"}}
+      {{#> form-group-label}}
+        {{#> form-label form-label--attribute=(concat 'for="' form--id form-group--id '"') required="true"}}Valid Redirect URIs{{/form-label}}
```
Comment (mattnolting) [inline src/patternfly/demos/Form/examples/BasicForms.md:317]:
```suggestion
        {{#> form-label form-label--attribute=(concat 'id="' form--id form-group--id '"') required="true"}}Valid Redirect URIs{{/form-label}}
```

File: src/patternfly/demos/Form/examples/BasicForms.md [hunk 2]
```diff
@@ -281,3 +263,96 @@ section: components
...
+        {{> form-group-label-help}}
+      {{/form-group-label}}
+      {{#> form-group-control form-group-control--modifier="pf-m-stack"}}
+        {{#> input-group}}
+          {{#> form-control controlType="input" input="true" form-control--attribute=(concat 'required type="text" id="' form--id form-group--id '-1" name="' form--id form-group--id '-1"')}}{{/form-control}}
```
Comment (mattnolting) [inline src/patternfly/demos/Form/examples/BasicForms.md:322]:
```suggestion
          {{#> form-control controlType="input" input="true" form-control--attribute=(concat 'required type="text" id="' form--id form-group--id '-input-1" name="' form--id form-group--id '-input-1" aria-labelledby="' form--id form-group--id ' ' form--id form-group--id '-input-1"')}}{{/form-control}}
```

File: src/patternfly/demos/Form/examples/BasicForms.md [hunk 3]
```diff
@@ -281,3 +263,96 @@ section: components
...
+          {{#> button button--modifier="pf-m-plain" button--attribute='aria-label="Remove"'}}
+            <i class="fas fa-minus-circle" aria-hidden="true"></i>
+          {{/button}}
+        {{/input-group}}
+        {{#> input-group}}
+          {{#> form-control controlType="input" input="true" form-control--attribute=(concat 'required type="text" id="' form--id form-group--id '-2" name="' form--id form-group--id '-2"')}}{{/form-control}}
```
Comment (mattnolting) [inline src/patternfly/demos/Form/examples/BasicForms.md:328]:
```suggestion
          {{#> form-control controlType="input" input="true" form-control--attribute=(concat 'required type="text" id="' form--id form-group--id '-input-2" name="' form--id form-group--id '-input-2" aria-labelledby="' form--id form-group--id ' ' form--id form-group--id '-input-2"')}}{{/form-control}}
```

File: src/patternfly/demos/Form/examples/BasicForms.md [hunk 4]
```diff
@@ -281,3 +263,96 @@ section: components
...
+          {{#> button button--modifier="pf-m-plain" button--attribute='aria-label="Remove"'}}
+            <i class="fas fa-minus-circle" aria-hidden="true"></i>
+          {{/button}}
+        {{/input-group}}
+        {{#> input-group}}
+          {{#> form-control controlType="input" input="true" form-control--attribute=(concat 'required type="text" id="' form--id form-group--id '-3" name="' form--id form-group--id '-3"')}}{{/form-control}}
```
Comment (mattnolting) [inline src/patternfly/demos/Form/examples/BasicForms.md:334]:
```suggestion
          {{#> form-control controlType="input" input="true" form-control--attribute=(concat 'required type="text" id="' form--id form-group--id '-input-3" name="' form--id form-group--id '-input-3" aria-labelledby="' form--id form-group--id ' ' form--id form-group--id '-input-3"')}}{{/form-control}}
```

**Review by mattnolting — COMMENTED** [review]
Looks great, left a few comments

File: src/patternfly/components/Form/form.scss [hunk 5]
```diff
@@ -230,6 +254,19 @@
     > * {
       margin-right: var(--pf-c-form__group-control--m-inline--child--MarginRight);
     }
+
+    > :last-child {
+      --pf-c-form__group-control--m-inline--child--MarginRight: 0;
+    }
+  }
+
+  &.pf-m-stack {
```
[the remark this answers is not in the document]
Comment (mcoker) [reply src/patternfly/components/Form/form.scss:263]: I considered that, but then you'd need to do math on elements with custom spacing (like helper text), where with flex you can just omit them from applying a gap... though everything else in the form uses grid and we're already doing that math other places (field groups), so I'm good with that, too.

[the remark this answers is not in the document]
Comment (mcoker) [reply src/patternfly/components/Form/form.scss:265]: @mattnolting updated

**Review by mattnolting — APPROVED** [review]
LPTM🤩

**Pull request merged**